src/test/java/de/wirvsvirus/testresult/backend/service/TestResultPushServiceTest.java

```diff
@@ -0,0 +1,105 @@
+
+import de.wirvsvirus.testresult.backend.model.TestResult;
+import de.wirvsvirus.testresult.backend.model.TestResult.Result;
+
+@ExtendWith(MockitoExtension.class)
+public class TestResultPushServiceTest {
+
+	@Mock
+	private SmsServiceProvider smsService;
+
+	@Mock
+	private EmailService emailService;
+
+	@InjectMocks
+	TestResultPushService service;
+
+	@Test
+	public void sendEmailForNegativeResult() {
+
+		TestResult tr = new TestResult();
+		tr.setContact("mail@test.com");
+		tr.setStatus(Result.NEGATIVE);
+		boolean sentFlag = service.executePush(tr);
+		assertAll("push is done and flag indicates it", () -> assertTrue(sentFlag),
+				() -> verify(emailService, times(1)).sendMail(any()),
+				() -> verify(smsService, never()).sendNegativeResultSms(any()));
+	}
+
+	@ParameterizedTest
+	@EnumSource(Result.class)
+	public void doNotSendEmailSentForNonNegativeResult(Result nonNegative) {
+
+		if (nonNegative.equals(Result.NEGATIVE))
+			return;
+
+		TestResult tr = new TestResult();
+		tr.setContact("mail@test.com");
+		tr.setStatus(nonNegative);
+		boolean sentFlag = service.executePush(tr);
+		assertAll("push is done and flag indicates it", () -> assertFalse(sentFlag),
+				() -> verify(emailService, never()).sendMail(any()),
+				() -> verify(smsService, never()).sendNegativeResultSms(any()));
+	}
+
+	@ParameterizedTest
+	@ValueSource(strings = { "004915555", "015555", "+491555" })
+	public void sendSmsForNegativeResultAndValidNumbers(String validNumber) {
+
+		TestResult tr = new TestResult();
+		tr.setContact(validNumber);
+		tr.setStatus(Result.NEGATIVE);
+		boolean sentFlag = service.executePush(tr);
+		assertAll("push is done and flag indicates it", () -> assertTrue(sentFlag),
+				() -> verify(emailService, never()).sendMail(any()),
+				() -> verify(smsService, times(1)).sendNegativeResultSms(any()));
+	}
+
+	@ParameterizedTest
+	@EnumSource(Result.class)
+	public void doNotSendSmsForNonNegativeResult(Result nonNegative) {
+
+		if (nonNegative.equals(Result.NEGATIVE))
+			return;
+
+		TestResult tr = new TestResult();
+		tr.setContact("017555");
+		tr.setStatus(nonNegative);
+		boolean sentFlag = service.executePush(tr);
+		assertAll("push is done and flag indicates it", () -> assertFalse(sentFlag),
+				() -> verify(emailService, never()).sendMail(any()),
+				() -> verify(smsService, never()).sendNegativeResultSms(any()));
+	}
+
+	@ParameterizedTest
+	@ValueSource(strings = { "+4855523", "07555", "foobar@bla", "foo.de", "-12314", "0047123123" })
+	public void doNotSendNotificationForInvalidContactPatterns(String invalidContact) {
+
+		TestResult tr = new TestResult();
+		tr.setContact(invalidContact);
+		tr.setStatus(Result.NEGATIVE);
+		boolean sentFlag = service.executePush(tr);
+		assertAll("push is done and flag indicates it", () -> assertFalse(sentFlag),
+				() -> verify(emailService, never()).sendMail(any()),
+				() -> verify(smsService, never()).sendNegativeResultSms(any()));
+	}

```

Comment:
Den assertAll-Block könnte man IMHO noch in eine methode auslagern (don't repeat yourself) - und der Test würde dadurch vermutlich noch besser lesbar